### ISSUE TITLE
Incorrect F811 when set/get property decorators are used

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#107]
In GitLab by @gforcada on Jun 17, 2015, 03:47

This snippet of code triggers a F811 while it shouldn't:

```python
class Example(object):
    my_data = None

    @getproperty
    def something(self):
        return self.my_data

    @setproperty
    def something(self, value):
        self.my_data = value
```

Is there any known workaround?

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jun 17, 2015, 06:17

Where do `@getproperty` and `@setproperty` come from?

PyFlakes only understands the following pattern (and allows for it):

```py
class Example(object):
    my_data = None

    @property
    def something(self):
        return self.my_data

    @something.setter
    def something(self, value):
        self.my_data = value
```

That is the canonical way of doing this in Python and it doesn't trigger F811.

**Comment by asottile** [comment]
In GitLab by @gforcada on Jun 20, 2015, 06:46

I see, we are using rwproperty... and now that I check its [PyPI page](https://pypi.python.org/pypi/rwproperty) it feels a bit old (last release on 2007...)

So no issue anymore, thanks for the prompt answer!

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Jun 20, 2015, 07:12

No worries! For what it's worth, looking at rwproperty, it was released for Python 2.4. The `property` descriptor and it's features above were introduced in 2.5 if I remember correctly. Hopefully this helps you (and others) ditch an unmaintained and now unneeded dependency. :smiley:
